XCode build fails

Hi, I'm trying to use this library and it works great in Android, but when I try to build for iOS (after applying the fix in the description) it fails with the following output:

With config.build_settings['SWIFT_VERSION'] = '4.1' in Podfile
 ```
Error output from Xcode build:
↳
    2018-11-26 19:04:07.144 xcodebuild[20997:3693078]  DVTAssertions: Warning in
    /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-13756/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/BuildSystem/Runtime/PB
    XTargetBuildContext.mm:757
    Details:  unexpected successful exit code from cancelled command <C0007:'CpHeader LocationPlugin.h':P8>
    Object:   <PBXTargetBuildContext: 0x7ff2a59d3f30>
    Method:   -createCommandInvocationRecordFromInvocation:
    Thread:   <NSThread: 0x7ff2a55bb360>{number = 5, name = (null)}
    Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/PermissionManager.swift:94:118: error: 'OpenExternalURLOptionsKey' is
    not a member type of 'UIApplication'
    fileprivate func convertToUIApplicationOpenExternalURLOptionsKeyDictionary(_ input: [String: Any]) -> [UIApplication.OpenExternalURLOptionsKey: Any] {
                                                                                                           ~~~~~~~~~~~~~ ^
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/PermissionManager.swift:55:77: error: type 'UIApplication' has no
    member 'openSettingsURLString'
                    let success = UIApplication.shared.openURL(URL.init(string: UIApplication.openSettingsURLString)!)
                                                                                ^~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-0.2.5+1/ios/Classes/FirebaseCorePlugin.m:13:17: warning: property 'dictionary' requires
    method 'dictionary' to be defined - use @dynamic or provide a method implementation in this category [-Wobjc-property-implementation]
    @implementation FIROptions (FLTFirebaseCorePlugin)
                    ^
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-0.2.5+1/ios/Classes/FirebaseCorePlugin.m:10:46: note: property declared here
    @property(readonly, nonatomic) NSDictionary *dictionary;
                                                 ^
    1 warning generated.
```

With config.build_settings['SWIFT_VERSION'] = '3.2' in Podfile
```
Xcode's output:
↳
    === BUILD TARGET FirebaseAuth OF PROJECT Pods WITH CONFIGURATION Debug ===
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/strategies/AudioVideoPermissionStrategy.swift:14:80: error: type
    'AVMediaType' (aka 'NSString') has no member 'video'
                return AudioVideoPermissionStrategy.getPermissionStatus(mediaType: AVMediaType.video)
                                                                                   ^~~~~~~~~~~ ~~~~~
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/strategies/AudioVideoPermissionStrategy.swift:16:80: error: type
    'AVMediaType' (aka 'NSString') has no member 'audio'
                return AudioVideoPermissionStrategy.getPermissionStatus(mediaType: AVMediaType.audio)
                                                                                   ^~~~~~~~~~~ ~~~~~
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/strategies/AudioVideoPermissionStrategy.swift:23:86: error:
    'AVMediaType' (aka 'NSString') is not implicitly convertible to 'String!'; did you mean to use 'as' to explicitly convert?
            let status: AVAuthorizationStatus = AVCaptureDevice.authorizationStatus(for: mediaType)
                                                                                         ^
                                                                                                   as String!
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/strategies/AudioVideoPermissionStrategy.swift:48:25: error: type
    'AVMediaType' (aka 'NSString') has no member 'video'
                mediaType = AVMediaType.video
                            ^~~~~~~~~~~ ~~~~~
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/strategies/AudioVideoPermissionStrategy.swift:50:25: error: type
    'AVMediaType' (aka 'NSString') has no member 'audio'
                mediaType = AVMediaType.audio
                            ^~~~~~~~~~~ ~~~~~
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/strategies/AudioVideoPermissionStrategy.swift:56:44: error:
    'AVMediaType' (aka 'NSString') is not implicitly convertible to 'String!'; did you mean to use 'as' to explicitly convert?
            AVCaptureDevice.requestAccess(for: mediaType, completionHandler: {
                                               ^
                                                         as String!
    === BUILD TARGET FirebaseAuth OF PROJECT Pods WITH CONFIGURATION Debug ===
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/PermissionManager.swift:94:118: error: 'OpenExternalURLOptionsKey' is
    not a member type of 'UIApplication'
    fileprivate func convertToUIApplicationOpenExternalURLOptionsKeyDictionary(_ input: [String: Any]) -> [UIApplication.OpenExternalURLOptionsKey: Any] {
                                                                                                           ~~~~~~~~~~~~~ ^
    /Users/nekothecat/devtools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/PermissionManager.swift:55:77: error: type 'UIApplication' has no
    member 'openSettingsURLString'
                    let success = UIApplication.shared.openURL(URL.init(string: UIApplication.openSettingsURLString)!)
                                                                                ^~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
```

Any help is greatly appreciated. Thanks for making permission handling easier for the rest of us! :)

- [x] :iphone: iOS
- [ ] :robot: Android

Can not build as well

> .pub-cache/hosted/pub.dartlang.org/permission_handler-2.1.1/ios/Classes/PermissionManager.swift:95:54: error: closure tuple parameter '(key: String, value: Any)' does not support destructuring
> return Dictionary(uniqueKeysWithValues: input.map { key, value in (UIApplication.OpenExternalURLOptionsKey(rawValue: key), value)})
> ^~~~~~~~~~
> (arg) -> <#Result#> let (key, value) = arg; return
> Could not build the application for the simulator.
> Error launching application on iPhone X.

--- a/example/ios/Podfile
+++ b/example/ios/Podfile
@@ -60,6 +60,7 @@ post_install do |installer|
   installer.pods_project.targets.each do |target|
     target.build_configurations.each do |config|
       config.build_settings['ENABLE_BITCODE'] = 'NO'
+      config.build_settings['SWIFT_VERSION'] = '4.2'
     end
   end
 end

--- a/example/ios/Runner.xcodeproj/project.pbxproj
+++ b/example/ios/Runner.xcodeproj/project.pbxproj
@@ -367,6 +367,7 @@
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
 				SDKROOT = iphoneos;
+				SWIFT_VERSION = 4.2;
 				TARGETED_DEVICE_FAMILY = "1,2";
 			};
 			name = Debug;
@@ -417,6 +418,7 @@
 				MTL_ENABLE_DEBUG_INFO = NO;
 				SDKROOT = iphoneos;
 				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
+				SWIFT_VERSION = 4.2;
 				TARGETED_DEVICE_FAMILY = "1,2";
 				VALIDATE_PRODUCT = YES;
 			};

--- a/ios/Classes/PermissionManager.swift
+++ b/ios/Classes/PermissionManager.swift
@@ -47,10 +47,14 @@ class PermissionManager: NSObject {
     static func openAppSettings(result: @escaping FlutterResult) {
         if #available(iOS 8.0, *) {
             if #available(iOS 10, *) {
-                UIApplication.shared.open(URL.init(string: UIApplication.openSettingsURLString)!, options: convertToUIApplicationOpenExternalURLOptionsKeyDictionary([:]),
-                                          completionHandler: {
-                                            (success) in result(success)
-                                          })
+                guard let url = URL(string: UIApplication.openSettingsURLString),
+                    UIApplication.shared.canOpenURL(url) else {
+                        return
+                }
+                
+                let optionsKeyDictionary = [UIApplication.OpenExternalURLOptionsKey(rawValue: "universalLinksOnly"): NSNumber(value: true)]
+                
+                UIApplication.shared.open(url, options: optionsKeyDictionary, completionHandler: nil)
             } else {
                 let success = UIApplication.shared.openURL(URL.init(string: UIApplication.openSettingsURLString)!)
                 result(success)

